### PR TITLE
Add Contravariant Basis Vector 2nd Derivatives

### PR DESCRIPTION
**Pull request by dpanici**
Adds 2nd derivatives of contravariant basis vectors

@FSaenzC needs these for his work on investigating the surface currents on surface of stellarators

code generated using this snippet, adapted from #474 

```python
import re
lab = ["rho", "theta", "zeta"]
tex = ["""{\\\\""" + foo + "}" for foo in lab]
cha = ["r", "t", "z"]
des = ["radial", "poloidal", "toroidal"]
Des = ["Radial", "Poloidal", "Toroidal"]

labels = {}

all_new = []

cnt = 0

for l in range(3):
    for i in range(3):
        for j in range(3):
            
            if cha[j] > cha[l]:
                continue
            
            cnt += 1
            
            temp_1st_deriv_1 = f'temp_{cha[j]} = cross(data["e_{lab[(i+1)%3]}_{cha[j]}"], data["e_{lab[(i+2)%3]}"]) + cross(data["e_{lab[(i+1)%3]}"], data["e_{lab[(i+2)%3]}_{cha[j]}"])'
            
            
            if l==j:
                # first and second variable are same, don't need to double-print the same derivative
                maybe_2nd = "2nd "
                which_coordinates = f"{des[j]} coordinate"
                temp_1st_deriv_2 = ""
            else:
                maybe_2nd=""
                which_coordinates = f"{des[j]} and {des[l]} coordinate"
                temp_1st_deriv_2 = f'temp_{cha[l]} =cross(data["e_{lab[(i+1)%3]}_{cha[l]}"], data["e_{lab[(i+2)%3]}"]) + cross(data["e_{lab[(i+1)%3]}"], data["e_{lab[(i+2)%3]}_{cha[l]}"])'
                
            e_bracketed_str = "{e}"
            units = "{-1}"
            s = (
f"""
@register_compute_fun(
    name="e^{lab[i]}_{cha[j]}{cha[l]}",
    label="\\\partial_{tex[j]}{tex[l]} \\\mathbf{e_bracketed_str}^{tex[i]}",
    units="m^{units}",
    units_long="inverse square meters",
    description="Contravariant {Des[i]} basis vector, {maybe_2nd}derivative"
     " wrt {which_coordinates}",
    dim=3,
    params=[],
    transforms={{}},
    profiles=[],
    coordinates="rtz",
    """
)
            src = (
f"""
def _e_sup_{lab[i]}_{cha[j]}{cha[l]}(params, transforms, profiles, data, **kwargs):
    temp = cross(data["e_{lab[(i+1)%3]}"], data["e_{lab[(i+2)%3]}"])
    {temp_1st_deriv_1}
    {temp_1st_deriv_2}
    temp_{cha[j]}{cha[l]} = cross(data["e_{lab[(i+1)%3]}_{cha[j]}{cha[l]}"], data["e_{lab[(i+2)%3]}"]) + cross(data["e_{lab[(i+1)%3]}_{cha[j]}"], data["e_{lab[(i+2)%3]}_{cha[l]}"]) + cross(data["e_{lab[(i+1)%3]}_{cha[l]}"], data["e_{lab[(i+2)%3]}_{cha[j]}"]) + cross(data["e_{lab[(i+1)%3]}"], data["e_{lab[(i+2)%3]}_{cha[j]}{cha[l]}"])
    
    data["e^{lab[i]}_{cha[j]}{cha[l]}"] = (temp_{cha[j]}{cha[l]}.T / data["sqrt(g)"] - (temp_{cha[j]}.T * data["sqrt(g)_{cha[l]}"] + temp_{cha[l]}.T * data["sqrt(g)_{cha[j]}"] + temp.T * data["sqrt(g)_{cha[j]}{cha[l]}"] )  / data["sqrt(g)"]**2 + 2 * temp.T * data["sqrt(g)_{cha[j]}"] * data["sqrt(g)_{cha[l]}"] /data["sqrt(g)"]**3).T
    return data
"""
            )
            pattern = r"data\[(.*?)\]"
            variables = re.findall(pattern, src)
            variables = [s.replace("'", "").replace('"', "") for s in variables]
            
            variables = sorted(list(set(variables)))
            all_new.append(variables[0])
            variables = sorted(list(set(variables)))[1:]
            
            s += f"data={variables}\n)"
            print(s+src)
```

### CODE REVIEW

**Comment by f0uriest** [comment]
Can you add tests for these, maybe something like https://github.com/PlasmaControl/DESC/blob/892beeaa0bc2e2667a3a516b6686235e738af0ed/tests/test_compute_funs.py#L1294

**Comment by dpanici** [comment]
Actually I need to sort out which have finite axis limits or not, and also update the pkl file for compute everything

**Comment by dpanici** [comment]
@unalmis the axis values of each of these new values seem to be `nan`, I just put them all as not implemented for now. I know some of the first derivatives like `e^rho_r` are defined at axis, so I bet `e^rho_rr` is also defined at axis, however I did not do the math out to implement the new lambda functions to evaluate them at axis. For now, these quantities are only needed at the surface anyways for Francisco's work, so I will leave that exercise to the future

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Changelog
 =========
-
+- Adds second derivatives of contravariant basis vectors to the data index.
```
Comment (f0uriest) [inline CHANGELOG.md:3]: I think "list of things we can compute" or something like that is better than "data index" since that's more internal to the code.

**Review by f0uriest — APPROVED** [review]

**Pull request merged**